// registration.proto
syntax = "proto3";

package registration;
import "google/protobuf/empty.proto";

message ServiceRegistration {
    string name = 1;
    string host = 2;
    int32 port = 3;
}

message DeregisterServiceRequest {
    string name = 1;
    string host = 2;
    int32 port = 3;
}

message SendServiceStatusRequest {
    string service_name = 1;
    int32 load = 2;
}

message Heartbeat {
    string service_name = 1;
}

service RegistrationService {
    rpc RegisterService(ServiceRegistration) returns (google.protobuf.Empty);
    rpc DeregisterService(DeregisterServiceRequest) returns (google.protobuf.Empty);
    rpc UpdateServiceStatus(SendServiceStatusRequest) returns (google.protobuf.Empty);
    rpc UpdateServiceHeartbeat(Heartbeat) returns (google.protobuf.Empty);
}